schema.graphql

```diff
@@ -1,45 +1,42 @@
-type NewOwnership @entity(immutable: true) {
-  id: Bytes!
-  from: Bytes! # address
-  to: Bytes! # address
-  blockNumber: BigInt!
-  blockTimestamp: BigInt!
-  transactionHash: Bytes!
+type GlobalState @entity {
+  id: ID!
+  oracles: [Oracle!]! @derivedFrom(field: "state")
+  activeOracles: [Oracle!]!
 }
 
-type NewPendingOwnership @entity(immutable: true) {
-  id: Bytes!
-  from: Bytes! # address
-  to: Bytes! # address
-  blockNumber: BigInt!
-  blockTimestamp: BigInt!
-  transactionHash: Bytes!
-}
-
-type OracleSet @entity(immutable: true) {
-  id: Bytes!
-  index: BigInt! # uint256
-  oracle: Bytes! # address
-  blockNumber: BigInt!
-  blockTimestamp: BigInt!
-  transactionHash: Bytes!
+type Oracle @entity {
+  id: Bytes! # address
+  index: String! # oracle_index
+  state: GlobalState!
+  latestConfig: OracleConfiguration!
+  configurations: [OracleConfiguration!]! @derivedFrom(field: "oracle")
+  votes: [OracleVote!]! @derivedFrom(field: "oracle")
+  active: Boolean!
+  activeSince: BigInt!
+  activeUntil: BigInt! # 0 means active
 }
 
 type OracleVote @entity(immutable: true) {
-  id: Bytes!
+  id: ID!
+  oracle: Oracle!
   subgraphDeploymentID: Bytes! # bytes32
   deny: Boolean! # bool
-  oracleIndex: BigInt! # uint256
   timestamp: BigInt! # uint256
-  blockNumber: BigInt!
-  blockTimestamp: BigInt!
-  transactionHash: Bytes!
 }
 
-type VoteTimeLimitSet @entity(immutable: true) {
-  id: Bytes!
-  voteTimeLimit: BigInt! # uint256
-  blockNumber: BigInt!
-  blockTimestamp: BigInt!
-  transactionHash: Bytes!
-}
+type OracleConfiguration @entity(immutable: true) {
+  id: ID!
+  oracle: Oracle!
+  commitHash: String!
+  ipfsConcurrency: String!
+  ipfsTimeout: String!
+  minSignal: String!
+  period: String!
+  gracePeriod: String!
+  supportedDataSourceKinds: String!
+  networkSubgraphId: String!
```

Comment:
Same comment about subgraph deployment id vs subgraph id

Reply:
Updated to use deployment id